WordPress/src/main/java/org/wordpress/android/util/DialogUtils.java

```diff
@@ -0,0 +1,56 @@
+package org.wordpress.android.util;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import org.wordpress.android.widgets.OpenSansEditText;
+import org.wordpress.android.widgets.WPTextView;
+
+public class DialogUtils {
+    public static void showMyProfileDialog(Context context, String title, String initialText, String hint, final Callback callback) {
+        LayoutInflater layoutInflater = LayoutInflater.from(context);
+        View promptView = layoutInflater.inflate(org.wordpress.android.R.layout.my_profile_dialog, null);
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(context);
+        alertDialogBuilder.setView(promptView);
+
+        final WPTextView textView = (WPTextView) promptView.findViewById(org.wordpress.android.R.id.my_profile_dialog_label);
+        final OpenSansEditText editText = (OpenSansEditText) promptView.findViewById(org.wordpress.android.R.id.my_profile_dialog_input);
+        final WPTextView hintView = (WPTextView) promptView.findViewById(org.wordpress.android.R.id.my_profile_dialog_hint);
+
+        textView.setText(title);
+        if (hint != null && !hint.isEmpty()) {
+            hintView.setText(hint);
+        }
+        else {
+            hintView.setVisibility(View.GONE);
+        }
+
+        if (initialText != null && !initialText.isEmpty()) {
+            editText.setText(initialText);
+            editText.setSelection(0, initialText.length());
+        }
+
+        alertDialogBuilder.setCancelable(false)
```

Comment:
Minor nit, feel free to ignore: I prefer `setCancelable(true)` so you can hit the back button to leave the dialog.